packages/flutter/lib/src/material/input.dart

```diff
@@ -521,26 +522,31 @@ class Input extends StatefulWidget {
 
 class _InputState extends State<Input> {
   final GlobalKey<_InputFieldState> _inputFieldKey = new GlobalKey<_InputFieldState>();
-  final GlobalKey _focusKey = new GlobalKey();
 
-  GlobalKey get focusKey => config.key is GlobalKey ? config.key : _focusKey;
+  FocusNode _focusNode;
+  FocusNode get _effectiveFocusNode => config.focusNode ?? (_focusNode ??= new FocusNode());
+
+  @override
+  void dispose() {
+    _focusNode?.dispose();
+    _focusNode = null;
```

Comment:
we don't generally null out fields in dispose; if it is important to do so here for some reason, please add a comment as to why